pkg/scalers/ibmmq_scaler.go

```diff
@@ -0,0 +1,218 @@
+	return nil
+}
+
+// parseIBMMQMetadata checks the existence of and validates the MQ connection data provided
+func parseIBMMQMetadata(config *ScalerConfig) (*IBMMQMetadata, error) {
+	meta := IBMMQMetadata{}
+
+	if val, ok := config.TriggerMetadata["host"]; ok {
+		_, err := url.ParseRequestURI(val)
+		if err != nil {
+			return nil, fmt.Errorf("invalid URL: %s", err)
+		}
+		meta.host = val
+	} else {
+		return nil, fmt.Errorf("no host URI given")
+	}
+
+	if val, ok := config.TriggerMetadata["queueName"]; ok {
+		meta.queueName = val
+	} else {
+		return nil, fmt.Errorf("no queue name given")
+	}
+
+	if val, ok := config.TriggerMetadata["queueDepth"]; ok && val != "" {
+		queueDepth, err := strconv.Atoi(val)
+		if err != nil {
+			return nil, fmt.Errorf("invalid targetQueueDepth - must be an integer")
+		} else {
+			meta.targetQueueDepth = queueDepth
+		}
+	} else {
+		fmt.Println("No target depth defined - setting default")
+		meta.targetQueueDepth = defaultTargetQueueDepth
+	}
+
+	if val, ok := config.TriggerMetadata["tls"]; ok {
+		tlsDisabled, err := strconv.ParseBool(val)
+		if err != nil {
+			return nil, fmt.Errorf("invalid tls setting: %s", err)
+		}
+		meta.tlsDisabled = tlsDisabled
+	} else {
+		fmt.Println("No tls setting defined - setting default")
+		meta.tlsDisabled = defaultTlsDisabled
+	}
+
+	if val, ok := config.AuthParams["username"]; ok {
+		meta.username = val
+	} else {
+		return nil, fmt.Errorf("no username given")
+	}
+
+	if val, ok := config.AuthParams["password"]; ok {
+		meta.password = val
+	} else {
+		return nil, fmt.Errorf("no password given")
+	}
+
+	return &meta, nil
+}
+
+// IsActive returns true if there are messages to be processed/if we need to scale from zero
+func (s *IBMMQScaler) IsActive(ctx context.Context) (bool, error) {
+	queueDepth, err := s.getQueueDepthViaHttp()
+	if err != nil {
+		return false, fmt.Errorf("error inspecting IBM MQ queue depth: %s", err)
+	}
+	return queueDepth > 0, nil
+}
+
+// getQueueDepthViaHttp returns the depth of the MQ Queue from the Admin endpoint
+func (s *IBMMQScaler) getQueueDepthViaHttp() (int, error) {
+
+	queue := s.metadata.queueName
+	url := s.metadata.host
+
+	var requestJson = []byte(`{"type": "runCommandJSON", "command": "display", "qualifier": "qlocal", "name": "` + queue + `", "responseParameters" : ["CURDEPTH"]}`)
+	req, err := http.NewRequest("POST", url, bytes.NewBuffer(requestJson))
+	req.Header.Set("ibm-mq-rest-csrf-token", "value")
+	req.Header.Set("Content-Type", "application/json")
+	req.SetBasicAuth(s.metadata.username, s.metadata.password)
+
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: s.metadata.tlsDisabled},
+	}
+	client := &http.Client{Transport: tr}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		panic(err)
+	}
+	defer resp.Body.Close()
+
+	body, _ := ioutil.ReadAll(resp.Body)
+
+	var response CommandResponse
+	json.Unmarshal(body, &response)
+
+	return response.CommandResponse[0].Parameters.Curdepth, nil
```

Comment:
this might panic if `CommandResponse` was empty, right?

Reply:
Thanks @ahmelsayed, we've included a potential fix for this in PR [#1259](https://github.com/kedacore/keda/pull/1259)